package/MDAnalysis/core/universe.py

```diff
@@ -494,8 +495,22 @@ def transfer_to_memory(self, start=None, stop=None, step=None,
                 coordinates = []  # TODO: use pre-allocated array
                 for i, ts in enumerate(self.trajectory[start:stop:step]):
                     coordinates.append(np.copy(ts.positions))
+                    dimensions.append(np.copy(ts.dimensions))
```

Comment:
annoyingly this will copy the box even for non NPT systems, but I guess it's a small cost compared to the number of atoms that will be floating around